### PR TITLE
Made Tests for Models

### LINKED ISSUE #23: Create Tests for Models
Create Unit tests for the Models

### PR DESCRIPTION
**Pull request by LokurasRlz**
# Testing

Created Unit Tests for Models using Rspec and Capybara

# Issues

- close #23

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -53,7 +53,8 @@ group :development, :test do
   gem 'debug', platforms: %i[mri mingw x64_mingw]
 
   # handle unit testing
-  gem 'rspec-rails'
+  gem 'capybara'
```
Comment (carlosmuerto) [inline Gemfile]: capybara is for page testing and is not designed for testing API

File: Gemfile.lock [hunk 2]
```diff
@@ -75,6 +75,15 @@ GEM
...
       msgpack (~> 1.2)
     builder (3.2.4)
+    capybara (3.38.0)
+      addressable
+      matrix
+      mini_mime (>= 0.1.3)
+      nokogiri (~> 1.8)
+      rack (>= 1.6.0)
+      rack-test (>= 0.6.3)
+      regexp_parser (>= 1.5, < 3.0)
```
Comment (carlosmuerto) [inline Gemfile.lock]: don't forget the bundle install after removing capybara

File: spec/models/appointment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+RSpec.describe Appointment, type: :model do
+  before(:each) do
+    @user = User.create(name: 'John Doe')
+    @doctor = Doctor.create(name: 'Dr. John Doe', specialization: 'Pediatrics', photo: 'https://www.google.com',
+                            user_id: @user.id)
+    @appointment = Appointment.create(datetime_of_appointment: '2024, 1, 1, 10, 00', description: 'Checkup',
+                                      user_id: @user.id, doctor_id: @doctor.id)
```
Comment (carlosmuerto) [inline spec/models/appointment_spec.rb]: Use some dynamically generate a time to ensure the test is agnostic of the testing date
 in this case you case use `1.years.from_now`

```ruby 
Appointment.create(
    user: seeded_user,
    doctor: seeded_doctor,
    description: "seeded description",
    datetime_of_appointment: 1.years.from_now
)
```

remember to apply this to the other cases

File: spec/models/appointment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+    it 'is valid with a time_of_appointment format of HH:MM' do
+      appointment = Appointment.new(user: @user, doctor: @doctor, description: 'abc',
+                                    datetime_of_appointment: '2024, 1, 1, 10, 00')
+      expect(appointment).to be_valid
+    end
+
+    it 'is not valid with a date_of_appointment in the past' do
+      appointment = Appointment.new(user: @user, doctor: @doctor, description: 'abc',
+                                    datetime_of_appointment: '2020-01-11 00:20:20.000000000 +0000')
```
Comment (carlosmuerto) [inline spec/models/appointment_spec.rb]: for this case you can use a negative number 

```ruby
appointment = Appointment.new(
    user: @user, 
    doctor: @doctor, 
    description: 'abc',
    datetime_of_appointment: -1.years.from_now
)
remember to apply this to the other cases

```

**Review by carlosmuerto — CHANGES_REQUESTED** [review]
# Status: Change Required 🟡 🟡 🟡 
### Hi @LokurasRlz 🙋‍♀️,  

## Highlights 🌞 

✔️  Good PR title and description
✔️  Commit messages are meaningful
✔️  No linter errors

✔️  ✔️  All test are well made

## Please check the comments below

**Comment by LokurasRlz** [comment]
@carlosmuerto, I have already applied the required changes.

**Review by carlosmuerto — APPROVED** [review]
# APPROVED 🟢 🟢 🟢

**Pull request merged**